### PR TITLE
Allow stable sort in TopK when sorting by indices

### PR DESCRIPTION
**Pull request by tomdol**
### Tickets:
 - 107467

### CODE REVIEW

**Review by mitruska — APPROVED** [review]
The change should be followed by corresponding test case for TopK-11 reference implementation.

File: docs/ops/sort/TopK_11.md [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 * *stable*
 
-  * **Description**: Specifies whether the equivalent elements should maintain their relative order from the input tensor. Takes effect only if the `sort` attribute is set to `value`.
+  * **Description**: Specifies whether the equivalent elements should maintain their relative order from the input tensor. Takes effect only if the `sort` attribute is set to `value` or `index`.
   * **Range of values**: `true` of `false`
   * **Type**: `boolean`
```
Comment (rkazants) [inline docs/ops/sort/TopK_11.md:36]: please update the specification below where you describe allowed pairs of `stable` and `sort` values and share that tricky example.

File: src/core/tests/type_prop/top_k.cpp [hunk 2]
```diff
@@ -435,5 +421,5 @@ TEST(type_prop, topk_v11_stable_sort_by_none) {
                                                                         element::i64,
                                                                         true),
                     NodeValidationFailure,
-                    HasSubstr("Stable sort can only be used when TopK's sorting mode is set to 'VALUE'"));
+                    HasSubstr("Stable sort can only be used when TopK's sorting mode is set to 'VALUE' or 'INDEX'"));
```
Comment (rkazants) [inline src/core/tests/type_prop/top_k.cpp:424]: can you please remove `xfail` for TF Top layer test: https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_TopKV2.py#L43

**Comment by tomdol** [comment]
> The change should be followed by corresponding update/test for TopK-11 reference implementation.

See: 108010

**Review by rkazants — APPROVED** [review]

**Review by PiotrKrzem — APPROVED** [review]

File: docs/ops/sort/TopK_11.md [hunk 3]
```diff
@@ -77,12 +77,39 @@ Sorting and minimum/maximum are controlled by `sort` and `mode` attributes with
   * *sort*=`value`, *mode*=`max`, *stable*=`true`  - descending by value, relative order of equal elements guaranteed to be maintained
   * *sort*=`value`, *mode*=`min`, *stable*=`false` - ascending by value, relative order of equal elements not guaranteed to be maintained
   * *sort*=`value`, *mode*=`min`, *stable*=`true`  - ascending by value, relative order of equal elements guaranteed to be maintained
-  * *sort*=`index`, *mode*=`max` - descending by index
-  * *sort*=`index`, *mode*=`min` - ascending by index
+  * *sort*=`index`, *mode*=`max`, *stable*=`false` - descending by index, relative order of equal elements not guaranteed to be maintained
```
Comment (xuchen-intel) [inline docs/ops/sort/TopK_11.md:80]: Please revert all the "descending by index" to be "ascending by index", as we've discussed in https://github.com/openvinotoolkit/openvino/pull/15583#discussion_r1133727732.

**Review by xuchen-intel — CHANGES_REQUESTED** [review]

File: docs/ops/sort/TopK_11.md (hunk 3 above)
Comment (tomdol) [reply docs/ops/sort/TopK_11.md]: Done, please have a look if the clarification in this commit https://github.com/openvinotoolkit/openvino/pull/16811/commits/a8e3ddad75a03d71c29dd278b0d1a2b4e56477e1 makes sense.

**Review by xuchen-intel — APPROVED** [review]

**Pull request merged**